core/tests/performance_framework/perf_services.py

```diff
@@ -58,21 +60,64 @@ class SeleniumPerformanceDataFetcher(object):
     # XHR requests made post initial page load will not be recorded.
     WAIT_DURATION_SECS = 3
 
-    def __init__(self, browser=DEFAULT_BROWSER_SOURCE):
+    BASE_URL = 'http://localhost:%d' % feconf.PERFORMANCE_TESTS_SERVER_PORT
+
+    LIBRARY_URL_SUFFIX = '/library'
+    EDITOR_URL_SLICE = '/create/'
+    PLAYER_URL_SLICE = '/explore/'
+    PROFILE_URL_SLICE = '/profile/'
+    LOGIN_URL_SUFFIX = '/_ah/login'
+    ADMIN_URL_SUFFIX = '/admin'
+    DASHBOARD_URL_SUFFIX = '/dashboard'
+    SPLASH_URL_SUFFIX = '/splash'
+    SIGNUP_URL_SUFFIX = '/signup'
+    DUMMY_URL = '/robots.txt'
+
+    def __init__(self, browser=DEFAULT_BROWSER_SOURCE, preload_options=None):
+        """preload_options:
+            login: True if login is required.
+            enable_explorations: True if explorations needs to be enabled.
+            enable_collections: True if collections needs to be enabled.
+            create_exploration: True if exploration needs to be created.
+        """
         if browser in self.SUPPORTED_BROWSER_SOURCES:
             self.browser = browser
         else:
             error_msg = 'Unsupported browser specified: %s' % browser
             raise ValueError(error_msg)
 
+        self.preload_options = preload_options
+        self.exploration_url = None
+        self.dev_appserver_login_cookie = None
+
+        random_id = random.randint(1, 100000)
+        self.email = 'test%d@example.com' % random_id
+        self.username = 'username%d' % random_id
+        self.username = self.preload_options.get('username', self.username)
+
+        driver = self._setup_driver()
+        if preload_options['login']:
+            self._setup_login(driver)
+        elif preload_options['enable_explorations']:
+            self._setup_enable_explorations(driver)
+        elif preload_options['enable_collections']:
+            self._setup_enable_collections(driver)
+        elif preload_options['create_exploration']:
+            self._setup_create_exploration(driver)
+        self._stop_driver(driver)
```

Comment:
Oh, I see, you're setting up an instance of the driver just to initialize stuff on the server, is that right?

I'm also wondering: do you know if it's these starts/stops of the driver that's causing the tests to take a really long time, or is it all the setup that's being done here, or is it something else?